api/v1beta1/azurecluster_default_test.go

```diff
@@ -24,8 +24,76 @@ import (
 	"github.com/Azure/go-autorest/autorest/to"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
 )
 
+func TestControlPlaneEndpointDefault(t *testing.T) {
+	cases := map[string]struct {
+		cluster *AzureCluster
+		output  *AzureCluster
+	}{
+		"set control plane endpoint host": {
+			cluster: &AzureCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "foo",
+				},
+				Spec: AzureClusterSpec{
+					ControlPlaneEndpoint: clusterv1.APIEndpoint{
+						Host: "apiserver.example.com",
+					},
+				},
+			},
+			output: &AzureCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "foo",
+				},
+				Spec: AzureClusterSpec{
+					ResourceGroup: "foo",
+					ControlPlaneEndpoint: clusterv1.APIEndpoint{
+						Host: "apiserver.example.com",
+					},
+				},
+			},
+		},
+		"set control plane endpoint port": {
+			cluster: &AzureCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "foo",
+				},
+				Spec: AzureClusterSpec{
+					ControlPlaneEndpoint: clusterv1.APIEndpoint{
+						Port: 8443,
+					},
+				},
+			},
+			output: &AzureCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "foo",
+				},
+				Spec: AzureClusterSpec{
+					ResourceGroup: "foo",
+					ControlPlaneEndpoint: clusterv1.APIEndpoint{
+						Port: 8443,
+					},
+				},
+			},
+		},
+	}
+
+	for name := range cases {
+		c := cases[name]
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			c.cluster.setResourceGroupDefault()
```

Comment:
why call setResourceGroupDefault default here? You should be able to add the test case to the existing `TestAzureCluster_ValidateUpdate` function in azurecluster_webhook_test.go

Reply:
Yeah, I'm still working through the testing code... I may need to add a webhook test as well I'm guessing?